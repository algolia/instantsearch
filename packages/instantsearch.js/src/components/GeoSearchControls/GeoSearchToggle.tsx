/** @jsx h */

import { h } from 'preact';

import type { ComponentChildren } from 'preact';

type Props = {
  classNameLabel: string;
  classNameInput: string;
  checked: boolean;
  onToggle: (event: Event) => void;
  children: ComponentChildren;
};

const GeoSearchToggle = ({
  classNameLabel,
  classNameInput,
  checked,
  onToggle,
  children,
}: Props) => (
  <label className={classNameLabel}>
    <input
      className={classNameInput}
      type="checkbox"
      checked={checked}
      onChange={onToggle}
    />
    {children}
  </label>
);

export default GeoSearchToggle;
