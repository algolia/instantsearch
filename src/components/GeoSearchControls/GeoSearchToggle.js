import React from 'preact-compat';
import PropTypes from 'prop-types';

const GeoSearchToggle = ({
  classNameLabel,
  classNameInput,
  checked,
  onToggle,
  children,
}) => (
  <label className={classNameLabel}>
    <input
      className={classNameInput}
      type="checkbox"
      checked={checked}
      onChange={onToggle}
    />
    {children}
  </label>
);

GeoSearchToggle.propTypes = {
  classNameLabel: PropTypes.string.isRequired,
  classNameInput: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onToggle: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default GeoSearchToggle;
