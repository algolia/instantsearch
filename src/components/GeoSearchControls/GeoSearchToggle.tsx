/** @jsx h */

import { h, ComponentChildren } from 'preact';

type Props = {
  classNameLabel: string;
  classNameInput: string;
  checked: boolean;
  onToggle(event: Event): void;
  children: ComponentChildren;
};

const GeoSearchToggle = ({
  classNameLabel,
  classNameInput,
  checked,
  onToggle,
  children,
}: Props) => (
  <label className={classNameLabel}>
    <input
      className={classNameInput}
      type="checkbox"
      checked={checked}
      onChange={onToggle}
    />
    {children}
  </label>
);

export default GeoSearchToggle;
