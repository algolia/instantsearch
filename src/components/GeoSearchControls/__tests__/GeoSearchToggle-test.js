/** @jsx h */

import { h } from 'preact';
import { shallow } from 'enzyme';
import GeoSearchToggle from '../GeoSearchToggle';

describe('GeoSearchToggle', () => {
  const defaultProps = {
    classNameLabel: 'label',
    classNameInput: 'input',
    checked: false,
    onToggle: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(
      <GeoSearchToggle {...props}>
        Clear the current map refinement
      </GeoSearchToggle>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render checked', () => {
    const props = {
      ...defaultProps,
      checked: true,
    };

    const wrapper = shallow(
      <GeoSearchToggle {...props}>
        Clear the current map refinement
      </GeoSearchToggle>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to call onToggle when the input changed', () => {
    const props = {
      ...defaultProps,
      onToggle: jest.fn(),
    };

    const wrapper = shallow(
      <GeoSearchToggle {...props}>
        Clear the current map refinement
      </GeoSearchToggle>
    );

    expect(props.onToggle).not.toHaveBeenCalled();

    wrapper.find('input').simulate('change');

    expect(props.onToggle).toHaveBeenCalled();
  });
});
