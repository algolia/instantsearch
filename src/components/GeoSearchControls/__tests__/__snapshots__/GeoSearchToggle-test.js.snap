// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoSearchToggle expect to render 1`] = `
<label
  className="label"
>
  <input
    checked={false}
    className="input"
    onChange={[Function]}
    type="checkbox"
  />
  Clear the current map refinement
</label>
`;

exports[`GeoSearchToggle expect to render checked 1`] = `
<label
  className="label"
>
  <input
    checked={true}
    className="input"
    onChange={[Function]}
    type="checkbox"
  />
  Clear the current map refinement
</label>
`;
